ipld/ipldsch/routing.ipldsch

```diff
@@ -0,0 +1,51 @@
+
+type ServiceEnvelope union {
+     | GetP2PProvideRequest "get-p2p-provide-request"
+     | GetP2PProvideResponse "get-p2p-provide-response"
+} representation keyed
+
+# TODO: IPLD schema type aliases not implemented
+# type Multihash = Bytes
+type Multihash struct {
+     Bytes Bytes
+}
+
+type GetP2PProvideRequest struct {
+     Keys [Multihash]
+}
+
+type ProvidersByKey struct {
+	Key Multihash
+	Provider Provider
+}
+
+type GetP2PProvideResponse struct {
+     # NOTE: ProvidersByKey {Multihash : Provider} is not possible in IPLD schema, instead:
+	ProvidersByKey [ProvidersByKey]
+}
+
+type Provider struct {
+     Node [Node]
+     Proto [TransferProtocol]
+     # can include other provider record meta here, e.g. the provenance of the provider record itself
+}
+
+# Node describes a node (identity, network address, authentication, whatever else applies)
+# We expect different types of nodes, e.g. peer, miner, public IP, etc.
+type Node union {
+     | Peer "peer"
+} representation keyed
+
+type Peer struct {
+     ID Bytes
+     Multiaddresses [Bytes]
+}
+
+type TransferProtocol union {
+     | BitswapTransfer "bitswap"
+} representation keyed
+
+type BitswapTransfer struct {}
```

Comment:
can we also [include](https://github.com/filecoin-project/index-provider/blob/main/metadata/metadata.go#L69-L77) / provide a representation for 
```
type FilecoinGraphsyncTransferV1 struct {
  PieceCID Link
  VerifiedDeal Bool
  FastRetrieval Bool
}
```